### ISSUE TITLE
Abstract class property readonly status in not enforced on subclasses

### ISSUE THREAD

**Issue by Ionaru** [microsoft/TypeScript#32193]
**TypeScript Version:**  3.6.0-dev.20190701

**Search Terms:** abstract, readonly, enforced, subclass

**Code**

```ts
abstract class Dog {
    public abstract readonly latinName: string;
}

class Corgi extends Dog {
    public latinName: string;

    constructor() {
        super();
        this.latinName = 'Canis lupus familiaris';
    }
}

const myDog = new Corgi();
myDog.latinName = 'Corgius Cuteness Supremus';
```

**Expected behavior:**
I expect the following error:
```
Class 'Corgi' incorrectly extends base class 'Dog'.
  Property 'latinName' is readonly in type 'Dog' but not in type 'Corgi'.
```
Like a private/public mismatch would give.

**Actual behavior:**
No error is given, class is extended without error and myDog.latinName is writable.

**Playground Link:** 
https://www.typescriptlang.org/play/#code/IYIwzgLgTsDGEAJYBthjAgIgewOYIG8AoBUhABwFcRkBLWBUSGeBKAU2ABNsA7ZAJ4JUEWrwBywALbsAXAmZjcAbiIBfIkRRoMAYWxRctBOwAeEdry4Yc+YmQrU6DEWMkz5i3is0PYfZkp4AwAKAEpCEgcyMEpydihw1WiyCAALWjAAOlcJaXYEAF4EAHJdYF5M4TjKDAAzaVo6YChMkuSyDQ0tAMQpAVsihF52AHcEfUNaJKJ+2xzgUTyZIbKDI1qJygsR9AQAZTiOKVr2oA

**Related Issues:** 
#11481, #11578, #14778 (all locked)

Most of these issues are from 2017, I'd like to get more information about why those issues were closed.
One claims this is gonna be implemented in the future, while another claims this is intended behaviour.

"Derived types are always allowed to have "more capability" than their base types.", why does this apply to readonly/writable but not to private/public ?

If this is indeed intended behaviour, then I'd expect a warning (or error?) discouraging the use of `abstract readonly propertyName` because it is the same as `abstract propertyName` and does not provide any additional enforcing.

**Comment by RyanCavanaugh** [comment]
> "Derived types are always allowed to have "more capability" than their base types.", why does this apply to readonly/writable but not to private/public ?

It applies to `protected`/`public`. Allowing derived types to make `private` fields `public` would defeat the purpose of `private` being a different keyword from `protected`.

> If this is indeed intended behaviour, then I'd expect a warning (or error?) discouraging the use of abstract readonly propertyName because it is the same as abstract propertyName and does not provide any additional enforcing.

This isn't correct reasoning. `abstract readonly propertyName` has effects on what you can do when working through an expression whose type is the base class.

**Comment by Ionaru** [comment]
> It applies to protected/public. Allowing derived types to make private fields public would defeat the purpose of private being a different keyword from protected.

I stand corrected.

> This isn't correct reasoning. abstract readonly propertyName has effects on what you can do when working through an expression whose type is the base class.

I did not consider using the abstract class as a type for anything. Thank you for taking the time to explain.
